core/src/main/java/org/apache/iceberg/jdbc/JdbcTableOperations.java

```diff
@@ -102,7 +102,7 @@ public void doRefresh() {
 
   @Override
   public void doCommit(TableMetadata base, TableMetadata metadata) {
-    String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+    String newMetadataLocation = writeNewMetadataIfRequired(base == null, metadata);
```

Comment:
Also to me, its clearer to add an extra line : 
```
boolean newTable = base == null;
String newMetadataLocation = writeNewMetadataIfRequired(newTable, metadata);
```

(if we keep the name I suggested above).